pkg/ipam/ipam.go

```diff
@@ -193,89 +193,143 @@ func (ipam *IPAM) AddOrUpdateSubnet(name, cidrStr, gw string, excludeIps []strin
-			subnet.V6Free = ips.Separate(subnet.V6Reserved)
-			subnet.V6Available = subnet.V6Free.Clone()
-			subnet.V6Using = subnet.V6Using.Intersect(ips)
-			subnet.V6Gw = v6Gw
-
-			pool := subnet.IPPools[""]
-			pool.V6IPs = ips
-			pool.V6Free = subnet.V6Free.Clone()
-			pool.V6Reserved = subnet.V6Reserved.Clone()
-			pool.V6Released = NewEmptyIPRangeList()
-			pool.V6Using = subnet.V6Using.Clone()
-
-			for name, p := range subnet.IPPools {
-				if name == "" {
-					continue
+
+		if protocol == kubeovnv1.ProtocolDual || protocol == kubeovnv1.ProtocolIPv6 {
+			if subnet.V6CIDR.String() != v6cidrStr || subnet.V6Gw != v6Gw {
+				_, cidr, _ := net.ParseCIDR(v6cidrStr)
+				subnet.V6CIDR = cidr
+				subnet.V6Reserved = v6Reserved
+				firstIP, _ := util.FirstIP(v6cidrStr)
+				lastIP, _ := util.LastIP(v6cidrStr)
+				ips, _ := NewIPRangeListFrom(fmt.Sprintf("%s..%s", firstIP, lastIP))
+				subnet.V6Free = ips.Separate(subnet.V6Reserved)
+				subnet.V6Available = subnet.V6Free.Clone()
+				subnet.V6Using = subnet.V6Using.Intersect(ips)
+				subnet.V6Gw = v6Gw
+
+				pool := subnet.IPPools[""]
+				pool.V6IPs = ips
+				pool.V6Free = subnet.V6Free.Clone()
+				pool.V6Reserved = subnet.V6Reserved.Clone()
+				pool.V6Released = NewEmptyIPRangeList()
+				pool.V6Using = subnet.V6Using.Clone()
+
+				for name, p := range subnet.IPPools {
+					if name == "" {
+						continue
+					}
+					p.V6Free = ips.Intersect(p.V6IPs)
+					p.V6Reserved = subnet.V6Reserved.Intersect(p.V6IPs)
+					p.V6Available = p.V6Free.Clone()
+					p.V6Released = NewEmptyIPRangeList()
+					pool.V6Free = pool.V6Free.Separate(p.V6IPs)
+					pool.V6Reserved = pool.V6Reserved.Separate(p.V6Reserved)
+				}
+				pool.V6Available = pool.V6Free.Clone()
+
+				for nicName, ip := range subnet.V6NicToIP {
+					if !ips.Contains(ip) {
+						podName := subnet.V6IPToPod[ip.String()]
+						klog.Errorf("%s address %s not in subnet %s new cidr %s", podName, ip.String(), name, cidrStr)
+						delete(subnet.V6NicToIP, nicName)
+						delete(subnet.V6IPToPod, ip.String())
+						continue
+					}
 				}
-				p.V6Free = ips.Intersect(p.V6IPs)
-				p.V6Reserved = subnet.V6Reserved.Intersect(p.V6IPs)
-				p.V6Available = p.V6Free.Clone()
-				p.V6Released = NewEmptyIPRangeList()
-				pool.V6Free = pool.V6Free.Separate(p.V6IPs)
-				pool.V6Reserved = p.V6Reserved.Separate(p.V6Reserved)
 			}
-			pool.V6Available = pool.V6Free.Clone()
-
-			for nicName, ip := range subnet.V6NicToIP {
-				if !ips.Contains(ip) {
-					podName := subnet.V6IPToPod[ip.String()]
-					klog.Errorf("%s address %s not in subnet %s new cidr %s", podName, ip.String(), name, cidrStr)
-					delete(subnet.V6NicToIP, nicName)
-					delete(subnet.V6IPToPod, ip.String())
-					continue
+			if !subnet.V6Reserved.Equal(v6Reserved) {
+				newAddV6Reserved := v6Reserved.Separate(subnet.V6Reserved)
+				// only support add reserved ip,because it will cause the ip repeated sometime when you add reserved ip and delete after sometime.
```

Comment:
```suggestion
				// only support add reserved ip, because it will cause the ip repeated sometime when you add reserved ip and delete after sometime.
```